File: 0005_Longest_Palindromic_Substring/Longest_Palindromic_Substring.cpp
class Solution {
private:
    int checkPalinLen(const string &s, int left, int right){
        int n = s.size();
        while(left >= 0 && right < n && s[left]==s[right]){
        left--;
        right++;
    }
    return right - left - 1;
};
public:
    string longestPalindrome(string s) {
        int n = s.size();

        if (n < 2) return s;
        int start = 0, maxLen = 0;
        for(int i=0; i < n; i++){
            int len1 = checkPalinLen(s, i, i);
            int len2 = checkPalinLen(s, i, i+1);
            int Len = max(len1, len2);
            if(maxLen < Len){
                maxLen = Len;
                start = i - (Len-1)/2;
            }
        }

        return s.substr(start, maxLen);
    }
};

File: 0005_Longest_Palindromic_Substring/Longest_Palindromic_Substring.py
# check C++ file for explaining comments

class Solution:
    def checkPalinLen(self, s, left, right):
        n = len(s)
        while(left >= 0 and right < n and s[left]==s[right]):
            left-= 1
            right+= 1
        
        return right - left - 1
    
    def longestPalindrome(self, s: str) -> str:
        n = len(s)

        if n < 2: return s
        start, maxLen = 0, 0
        for i in range(0, n):
            len1 = self.checkPalinLen(s, i, i)
            len2 = self.checkPalinLen(s, i, i+1)
            Len = max(len1, len2)
            if (maxLen < Len):
                maxLen = Len
                start = i - (Len-1)//2
        
        return s[start: start + maxLen]